docs/embedding/sdk/questions.md

```diff
@@ -69,15 +69,14 @@ export default function App() {
 | height             | number or string                                                     | (optional) A number or string specifying a CSS size value that specifies the height of the component                                                                                                                                                                                                               |
 | entityTypeFilter   | string array; options include "table", "question", "model", "metric" | (optional) An array that specifies which entity types are available in the data picker                                                                                                                                                                                                                             |
 | isSaveEnabled      | boolean                                                              | (optional) Whether people can save the question.                                                                                                                                                                                                                                                                   |
 | withResetButton    | boolean                                                              | (optional, default: `true`) Determines whether a reset button is displayed. Only relevant when using the default layout                                                                                                                                                                                            |
-| withTitle          | boolean                                                              | (optional, default: `false`) Determines whether the question title is displayed. Only relevant when using the default layout.                                                                                                                                                                                      |
-| customTitle        | string or undefined                                                  | (optional) Allows a custom title to be displayed instead of the default question title. Only relevant when using the default layout.                                                                                                                                                                               |
+| title              | boolean or string or `ReactNode` or `() => ReactNode`                | (optional) Determines whether the question title is displayed, and allows a custom title to be displayed instead of the default question title. Shown by default. Only relevant when using the default layout.                                                                                                     |
```

Comment:
@WiNloSt Good point. I think it might have to do with how static question and interactive question are used differently; today we use static questions for simply displaying questions itself, while interactive question needs a more complex layout that supports the querying experience.

I have clarified this [in the docs](https://github.com/metabase/metabase/pull/50660/commits/771c62509ff81c982c1241755e9511e1e8ac21cb) for the time being that `title` only applies to static questions.

![CleanShot 2567-11-30 at 02 44 19@2x](https://github.com/user-attachments/assets/39dcdfe7-3c00-407e-b308-245495bba5e8)

I'll check @albertoperdomo and @oisincoveney - do you think it makes sense to add titles to static questions for consistency?

Reply:
@WiNloSt Sounds good! I opened a Slack thread on this to discuss the feature parity more in detail. I agree with supporting `title`.

In my current mental model static components isn't supposed to support interactivity though, so `<StaticQuestion.Filter />` (or even dashboard filters in static dashboards) sounds a bit out-of-place to me.

Reply:
Well, that's a fair statement. Maybe try to think less about the actual component being used, but in terms of customizing the layout. If anything's applicable to static questions that we can use from interactive questions. I'd suggest we give the static ones the same treatment e.g. `StaticQuestion.Whatever`. This allows users to customize the layout to their liking. Although, currently it doesn't seem we have a bunch of sub components that could work on the static embeds. I think `StaticEmbed.Title` could work, but that might be the only one. So, it might not be tempting to support compound components and custom layouts. But for consistency, and comprehensibility I'd still suggest we do it (if it is not too hard)